vortex-dtype/src/field.rs

```diff
@@ -59,7 +63,19 @@ impl Field {
     }
 }
 
-/// A path through a (possibly nested) struct, composed of a sequence of field selectors
+/// A sequence of field selectors representing a path through zero or more layers of `DType`.
+///
+/// # Examples
+///
+/// The empty path references the root:
+///
+/// ```
+/// use vortex_dtype::*;
+///
+/// let i32 = DType::Primitive(PType::I32, Nullability::NonNullable);
```

Comment:
I didn't realize this was even valid syntax. Can we call it i32_col or something. Idk why but it's jarring to me :P

Reply:
Done.